src/js/bootstrap-switch.js

```diff
@@ -394,22 +394,38 @@ class BootstrapSwitch {
       this.setPrevOptions()
       this._width()
       this._containerPosition()
+
+      // Allow browser repaint to end of event queue but minimize flicker
       setTimeout(() => {
+        // Ensure visible in case hidden while calculating widths
+        this.$wrapper.css('visibility', 'visible')
         if (this.options.animate) {
           return this.$wrapper.addClass(this._getClass('animate'))
         }
-      }, 50)
+      }, 0)
     }
+
+    // initialize if the $wrapper is already visible
     if (this.$wrapper.is(':visible')) {
       init()
       return
     }
-    const initInterval = window.setInterval(() => {
+
+    let interval = 0.5
```

Comment:
No need for let:

```js
const poll = (check, interval = 0.5) => {
   if (!check()) {
       setTimeout(() => poll(check, interval * 2), interval);
   }
}
```
Then use it:
```js
poll(() => (this.$wrapper.is(':visible') && (init() || true)));
```

Although you'll need to port it to 'standard JS'